### PR TITLE
HUMID: Add cluster statistics

### PR DESCRIPTION
**Pull request by Redmar-van-den-Berg**
- Split up the parsing of HUMID statistics into independent files
- Add three additional plots, which show clustering statistics from HUMID
- Rename the "M Unique" column in General Statistics to "M Unique Reads", to match UMI-Tools, which is a similar tool
- Add the % of reads after deduplication to the General Statistics table, to match UMI-Tools

### CODE REVIEW

**Comment by vladsavelyev** [comment]
Thanks a lot @Redmar-van-den-Berg for the PR! 

I wonder if you happen to have test files `neigh.dat`, `counts.dat`, `clusters.dat` to put into https://github.com/MultiQC/test-data/tree/main/data/modules/humid?

File: multiqc/modules/humid/neighbours.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    plot_config = {
+        "id": "humid-neighbours",
+        "title": "HUMID: Neighbour statistics",
+        "ylab": "Number of sequences",
+        "xlab": "Number of neighbours",
+        "logswitch": True,
+        "logswitch_active": True,
+        "hide_zero_cats": False,
+    }
+    self.add_section(
```
Comment (vladsavelyev) [inline multiqc/modules/humid/neighbours.py:57]: If neigh.dat is not found, an empty plot will be generated. Would be nice to check if `self.neighbours` is non-empty before the `self.add_section` call

**Comment by Redmar-van-den-Berg** [comment]
I've created a pull request with some example data here: https://github.com/MultiQC/test-data/pull/313

File: multiqc/modules/humid/neighbours.py (hunk 1 above)
Comment (Redmar-van-den-Berg) [reply multiqc/modules/humid/neighbours.py:57]: Each submodule separately searches for their associated log file, so if a file is completely missing it will not be included in the report.
The tool for this module (HUMID) will never generate empty statistics files.

Comment (Redmar-van-den-Berg) [reply multiqc/modules/humid/neighbours.py:57]: I did some more testing, and you can indeed add up with an empty section if there is only a single sample, and that sample cluster file has no data. I've fixed this in the latest commit, thanks!

**Review by vladsavelyev — APPROVED** [review]
Looks great! Minor things which I pushed directly into the branch, if you don't mind - and happy to merge 👍

**Comment by ewels** [comment]
@multiqc-bot changelog

**Comment by vladsavelyev** [comment]
Ah, figured why the tests were failing. `self.cluster` vs `self.clusters` were interchanged in the code

**Review by ewels — APPROVED** [review]
Great stuff, thanks!
